services/pdf-ms/helpers/pdf.ts

```diff
@@ -50,31 +50,40 @@ const replaceTextInTextObject = (textObject: TextObject, json: Record<string, an
   const templateStrings = [...textObject.text.matchAll(regex)];
   const newTextObject: TextObject = { ...textObject };
 
-  const newText = templateStrings.reduce((prev, currentRegexResult) => {
+  const newText = templateStrings.reduce((previous, currentRegexResult) => {
     const replacement = getPropertyFromDottedString(json, currentRegexResult[1].trim());
```

Comment:
what is [1]?

Reply:
So these results come from the [matchAll function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll), which finds all regex matches and returns an array of results, where each result has the form of another array of the form [ fullMatch, capturingGroup1, capturingGroup2, ...]. So in this case, I just have one capturing group in my regex, so [1] gets that, while [0] gets the full expression, including the {{ }}.

Reply:
Suggestion here is to destruct the array values, might be easier to understand then.
```const [ fullMatch, capturingGroup1, capturingGroup2, ...] = currentRegexResult```

Reply:
yeah, sounds like a good idea for readability, I'll do that.